### PR TITLE
Add arrays syntax

### PR DESCRIPTION
**Pull request by rkilar**
I noticed that the `type` field in `ArrayAllocation` was of type `String` instead of `Type`, which I missed when reviewing the PR.

### CODE REVIEW

File: app/src/test/kotlin/compiler/e2e/CorrectProgramsE2eTest.kt
Comment (ignacy123) [inline app/src/test/kotlin/compiler/e2e/CorrectProgramsE2eTest.kt:256]: Please add two more tests before this - one per allocation type. It's gonna make it more obvious which allocation type is not implemented correctly during debugging

File: app/src/test/kotlin/compiler/syntax/AstFactoryTest.kt
Comment (ignacy123) [inline app/src/test/kotlin/compiler/syntax/AstFactoryTest.kt:728]: Please add a test for the one value allocation type

**Review by ignacy123 — CHANGES_REQUESTED** [review]
We'll merge this after the other parts, but everything is looking good for me :+1:

File: app/src/main/kotlin/compiler/syntax/LanguageTokens.kt
Comment (bebidek) [inline app/src/main/kotlin/compiler/syntax/LanguageTokens.kt:41]: Are you sure that "alokacja" is the best choice here? While it is a Polish word, it's a loan translation from English "allocation" while we tend to prefer words of more Polish origin (look for example at "czynność" instead of "funkcja").

Comment (rkilar) [reply app/src/main/kotlin/compiler/syntax/LanguageTokens.kt:41]: Yeah, I wasn't so sure about it myself. I used it to match the string representation in ASTNode. I think "przydział" is the best Polish alternative. We can also use word not related to "allocate" but maybe "new", "create", "array" or "table". I need some outside opinion on that.

Comment (ignacy123) [reply app/src/main/kotlin/compiler/syntax/LanguageTokens.kt:41]: Maybe just "tablica"?

Comment (bebidek) [reply app/src/main/kotlin/compiler/syntax/LanguageTokens.kt:41]: Or "ciąg" if we prefer unusual keywords.

Comment (rkilar) [reply app/src/main/kotlin/compiler/syntax/LanguageTokens.kt:41]: It could also be "rząd", "szyk" (this one technically has German etymology) or "szereg" (from Hungarian).

**Review by ignacy123 — APPROVED** [review]

**Comment by rkilar** [comment]
I had some problems with missing cases in type checking. I think I fixed them, as the end to end tests now fail on unimplemented cases in InitializationVerifier, but it should probably be checked.

**Review by ignacy123 — CHANGES_REQUESTED** [review]
I also came across these problems, and already prepared a [new branch](https://github.com/TCS-Compilers/compiler/tree/fix-type-checking) with fixes.
Since you already started applying the fixes here, can you unify my work and yours before merging? I found one more issue in name resolution and added a new test to type checking (though you will have to adjust the final assert, as you are raising a different exception).

**Review by ignacy123 — APPROVED** [review]

**Pull request merged**
